gamemodes/terrortown/entities/weapons/weapon_randomat_christmas_cannon.lua

```diff
@@ -0,0 +1,130 @@
+AddCSLuaFile()
+
+if CLIENT then
+    SWEP.PrintName = "Christmas Cannon"
+    SWEP.Slot = 8
+end
+
+SWEP.Base = "weapon_tttbase"
+SWEP.HoldType = "rpg"
+SWEP.DrawAmmo = false
+SWEP.DrawCrosshair = false
+SWEP.Weight = 5
+
+SWEP.Primary.Automatic = false
+SWEP.Primary.Ammo = ""
+SWEP.Primary.Delay = 1
+SWEP.Primary.Recoil = 50
+SWEP.Primary.Damage = 0
+SWEP.Primary.ClipSize = -1
+SWEP.Primary.ClipMax = -1
+SWEP.Primary.DefaultClip = -1
+
+SWEP.ViewModelFlip = false
+SWEP.ViewModelFOV = 55
+SWEP.ViewModel = Model("models/weapons/v_rpg.mdl")
+SWEP.WorldModel = Model("models/weapons/w_rocket_launcher.mdl")
+
+SWEP.Kind = WEAPON_ROLE
+SWEP.AutoSpawnable = false
+SWEP.AmmoEnt = "none"
+SWEP.AllowDrop = false
+SWEP.IsSilent = false
+SWEP.NoSights = true
+
+local ShootSound = Sound("weapons/grenade_launcher1.wav")
+
+function SWEP:Initialize()
+    self:SetWeaponHoldType(self.HoldType)
+    if CLIENT then
+        self:AddHUDHelp("Use {primaryfire} to give gifts to nice children", "Use {secondaryfire} to shoot coal at naughty children", false)
+    end
+end
+
+function SWEP:PrimaryAttack()
+    if not IsFirstTimePredicted() then return end
+    self:SetNextPrimaryFire(CurTime() + self.Primary.Delay)
+    self:SetNextSecondaryFire(CurTime() + self.Primary.Delay)
+
+    local owner = self:GetOwner()
+    if not IsValid(owner) then return end
+
+    local has_ammo = owner:GetNWBool("XmasCannonHasAmmo", false)
+
+    if has_ammo then
+        self:SendWeaponAnim(ACT_VM_PRIMARYATTACK)
+        self:EmitSound(ShootSound)
+
+        if SERVER then
+            owner:SetNWBool("XmasCannonHasAmmo", false)
+
+            local roles = { ROLE_TRAITOR, ROLE_DETECTIVE }
```

Comment:
This can be replaced with a call to
```suggestion
            local roles = { ROLE_TRAITOR, ROLE_DETECTIVE }
            local tracking = 0
            local _, _, swep_table = Randomat:GetShopEquipment(nil, roles, nil, false, tracking, function(value) tracking = value end, true)
            local item_id = WEPS.GetClass(swep_table)
```
The first `nil` is because the player is only required when getting equipment and since we're looking for droppables only, equipment does not apply.
The second `nil` is because there is no blocklist.
The `false` is to exclude equipment, though that is inherent because of the `true` at the end which means only droppable things are included.